src/internet/subsonic/subsonicdynamicplaylist.cpp

```diff
@@ -0,0 +1,259 @@
+/* This file is part of Clementine.
+   Copyright 2015, Nick Lanham <nick@afternight.org>
+
+   Clementine is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Clementine is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Clementine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "subsonicdynamicplaylist.h"
+
+#include <QEventLoop>
+#include <QFileInfo>
+#include <QSslConfiguration>
+#include <QXmlStreamReader>
+
+#include "core/application.h"
+#include "core/logging.h"
+#include "core/network.h"
+#include "core/taskmanager.h"
+#include "core/timeconstants.h"
+#include "internet/core/internetplaylistitem.h"
+
+SubsonicDynamicPlaylist::SubsonicDynamicPlaylist()
+  : stat_(QueryStat_Newest),
+    offset_(0) {}
+
+SubsonicDynamicPlaylist::SubsonicDynamicPlaylist(const QString& name,
+                                                 QueryStat stat)
+  : stat_(stat),
+    offset_(0) {
+  set_name(name);
+}
+
+void SubsonicDynamicPlaylist::Load(const QByteArray& data) {
+  QDataStream s(data);
+  s >> *this;
+}
+
+void SubsonicDynamicPlaylist::Load(QueryStat stat) {
+  stat_ = stat;
+}
+
+QByteArray SubsonicDynamicPlaylist::Save() const {
+  QByteArray ret;
+  QDataStream s(&ret, QIODevice::WriteOnly);
+  s << *this;
+  return ret;
+}
+
+// copied from SubsonicService
+QNetworkReply* SubsonicDynamicPlaylist::Send(QNetworkAccessManager& network, const QUrl& url, const bool usesslv3) {
+  QNetworkRequest request(url);
+  // Don't try and check the authenticity of the SSL certificate - it'll almost
+  // certainly be self-signed.
+  QSslConfiguration sslconfig = QSslConfiguration::defaultConfiguration();
+  sslconfig.setPeerVerifyMode(QSslSocket::VerifyNone);
+  if (usesslv3) {
+    sslconfig.setProtocol(QSsl::SslV3);
+  }
+  request.setSslConfiguration(sslconfig);
+  QNetworkReply* reply = network.get(request);
+  return reply;
+}
+
+PlaylistItemList SubsonicDynamicPlaylist::Generate() { return GenerateMore(10); }
+
+PlaylistItemList SubsonicDynamicPlaylist::GenerateMore(int count) {
+  SubsonicService* service = InternetModel::Service<SubsonicService>();
```

Comment:
I don't think so, because these lists get serialized and then reconstructed in Generator (generator.cpp line 40).  I could maybe put the singleton call there, but it's not clear that would be any better.  Let me know if you'd prefer that and I can see about getting it to work.